src/desktop/apps/auction/components/DOM.js

```diff
@@ -64,12 +86,10 @@ class DOM extends Component {
 
       // If the user already has a CC, show accept conditions
       // (which redirects to auction-registration/:slug)
-    } else if (me.has_credit_cards) {
-      this.showAcceptConditions()
-
-      // Redirect to credit card registration form
-    } else {
+    } else if (!me.has_credit_cards) {
       window.location.assign(auction.registerUrl())
+    } else {
+      this.showModal("RegistrationFlow")
```

Comment:
`handleRegister()` is triggered by the clicking the register button or the presence of `/registration-flow` in the url.
`handleRegister` may take a few alternate paths to registration:
- via new modal
- via auth with redirect to registration flow route
- by sending the user to the credit card form registration url

The latter 2 of these should happen before `maybeShowModal()` automatically executes but there may be a nicer way to clean up the reasoning here, if only for us people.